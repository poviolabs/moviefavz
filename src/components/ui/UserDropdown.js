import React from 'react';
import PropTypes from 'prop-types';

import { NavLink } from 'react-router-dom';

import styled from 'styled-components';

import { Dropdown, Avatar, Menu, Space } from 'antd';
import { UserOutlined, DownOutlined } from '@ant-design/icons';

const StyledDropdownLink = styled.a`
  color: ${({ theme }) => theme.colors.text};

  &:hover {
    color: ${({ theme }) => theme.colors.primary};
  }
`;

const UserDropdown = ({ user, onLogout }) => {
  return (
    <Dropdown
      overlay={
        <Menu>
          <Menu.Item>
            <NavLink to="/your-app-usage">Your app usage</NavLink>
          </Menu.Item>
          <Menu.Item>
            <span onClick={onLogout}>Log out</span>
          </Menu.Item>
        </Menu>
      }
      trigger={['click']}
    >
      <StyledDropdownLink
        className="ant-dropdown-link"
        onClick={(e) => e.preventDefault()}
      >
        <Space>
          <Avatar
            size="small"
            src={user.picture}
            icon={!user.picture ? <UserOutlined /> : null}
          />
          <span>Hi, {user.given_name || user.nickname}</span>
          <DownOutlined />
        </Space>
      </StyledDropdownLink>
    </Dropdown>
  );
};

UserDropdown.propTypes = {
  user: PropTypes.object,
  onLogout: PropTypes.func,
};

export default UserDropdown;
