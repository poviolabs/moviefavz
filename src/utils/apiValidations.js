const validResponse = (status) => status && status >= 200 && status < 300;

export default { validResponse };
