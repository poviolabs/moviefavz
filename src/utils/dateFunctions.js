import dayjs from 'dayjs';

const getUnixDay = (timestamp = Date.now()) => {
  return dayjs(timestamp).startOf('date').valueOf();
};

const fillDaysToMatch = (num, data) => {
  if (!data.length) {
    return [];
  }
  const [lastDate] = data.reverse();

  return Array.from('x'.repeat(num - data.length))
    .map((_, i) => i + 1)
    .reverse()
    .map((i) => {
      return {
        value: 0,
        timestamp: dayjs(lastDate.timestamp).subtract(i, 'day').valueOf(),
      };
    });
};

export default { getUnixDay, fillDaysToMatch };
