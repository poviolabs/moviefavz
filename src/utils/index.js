export { default as arrayFunction } from './arrayFunctions';
export { default as apiValidations } from './apiValidations';
