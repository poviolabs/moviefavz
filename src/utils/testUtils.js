import React from 'react';
import { render } from '@testing-library/react';

import { BrowserRouter } from 'react-router-dom';

import { ThemeProvider } from 'styled-components';
import { theme } from '../styles';

// eslint-disable-next-line react/prop-types
const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={theme}>
      <BrowserRouter>{children}</BrowserRouter>
    </ThemeProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
