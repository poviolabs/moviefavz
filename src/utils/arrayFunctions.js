import dateFunctions from './dateFunctions';

const groupArr = (data, size) => {
  const group = [];
  for (let i = 0, j = 0; i < data.length; i++) {
    if (i >= size && i % size === 0) {
      j++;
    }
    group[j] = group[j] || [];
    group[j].push(data[i]);
  }
  return group;
};

const groupByDay = (data, key = 'timestamp') => {
  const groupedByKey = data.reduce((obj, item) => {
    const timestamp = item[key];
    const day = dateFunctions.getUnixDay(timestamp);

    if (Object.prototype.hasOwnProperty.call(obj, day)) {
      obj[day].value += 1;
    } else {
      obj[day] = {
        value: 1,
        timestamp,
      };
    }
    return obj;
  }, {});
  return Object.values(groupedByKey);
};

export default { groupArr, groupByDay };
