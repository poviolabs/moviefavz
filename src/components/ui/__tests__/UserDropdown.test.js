import React from 'react';
import { render } from '../../../utils/testUtils';

import UserDropdown from '../UserDropdown';

describe('<UserDropdown />', () => {
  it('Renders the Component', () => {
    const props = {
      user: {},
      onLogout: jest.fn(),
    };
    const container = render(<UserDropdown {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
