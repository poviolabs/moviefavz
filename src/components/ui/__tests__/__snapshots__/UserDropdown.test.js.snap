// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserDropdown /> Renders the Component 1`] = `undefined`;
